baseapp/baseapp.go

```diff
@@ -288,10 +288,13 @@ func (app *BaseApp) storeConsensusParams(consensusParams *abci.ConsensusParams)
 }
 
 // getMaximumBlockGas gets the maximum gas from the consensus params.
-func (app *BaseApp) getMaximumBlockGas() (maxGas uint64) {
-	if app.consensusParams == nil || app.consensusParams.BlockSize == nil {
+func (app *BaseApp) getMaximumBlockGas() uint64 {
+	if app.consensusParams == nil ||
+		app.consensusParams.BlockSize == nil ||
+		app.consensusParams.BlockSize.MaxGas <= 0 {
```

Comment:
Should we check exactly `== -1` and panic for other negative numbers?

Reply:
(it seems confusing to accept any negative number when the semantics don't change)

Reply:
Yes, good catch!

Reply:
Updated!